assets/app/scripts/services/data.js

```diff
@@ -42,7 +42,7 @@ angular.module('openshiftConsole')
     for (var i = 0; i < subAttrs.length; i++) {
       attrValue = attrValue[subAttrs[i]];
       if (!attrValue) {
-        return;
+        attrValue = "";
```

Comment:
I'm pretty sure I don't even need this change anymore.  What would you expect it to return in this case, if for the current object it doesn't have anything for the given attribute.  Just return like it was before so the object will not be added to the map that was passed in?

By Setting attrValue to "" it allows the secondaryAttr to still be applied.  Maybe it should _only_ do the "" behavior if secondaryAttr is passed in.